### ISSUE TITLE
crates.io takes ~6 seconds to load

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/crates.io#1304]
Crates.io used to load almost immediately, but now seems to consistently take around 6 seconds to fully load.  This includes spinners like this:

![image](https://user-images.githubusercontent.com/547158/38257800-1717c8c8-37b5-11e8-8e64-deabb6ae7c78.png)

Not sure if it's just for some people (maybe it's a NZ thing?)

**Comment by carols10cents** [comment]
Does this happen in all browsers? Can you grab a dump of the network tab from the developer tools to see what's hanging exactly?

**Comment by OvermindDL1** [comment]
Loading the css and JS is also taking over 700ms here.  The summary API that @aturon linked is taking 5.52 *seconds* here.  `crates.io` itself took ~500ms.  The images are taking 200-500ms.  In comparison, other sites are loading fine at 20-200ms such as google.com, my personal website hosted 1000 miles away, etc...

**Comment by sgrif** [comment]
I was able to reproduce just a minute ago, but that is no longer the case.

**Comment by OvermindDL1** [comment]
It seems to bounce I've noticed, sometimes speedy fast, sometimes dreadfully slow, it's quite random.

**Comment by carols10cents** [comment]
I have not deployed lately. I don't see any incidents on https://status.heroku.com/. There may be database queries that could use some tweaking, i'm going to investigate that next.

**Comment by OvermindDL1** [comment]
Even the css/js is slowed down at times as well though.

Ahh, it's heroku, I never see servers working well on that service...

**Comment by carols10cents** [comment]
> Ahh, it's heroku, I never see servers working well on that service...

@OvermindDL1 I assure you there are quite a few websites you use that serve at large scales that work perfectly well on heroku; generalizations like that aren't welcome here.

**Comment by OvermindDL1** [comment]
Oh sorry, didn't mean it quite that way!  Just bad experiences with a number of other sites on the same service...  ^.^;

**Comment by sivakumar-kailasam** [comment]
Took me more than a few seconds earlier today and when i checked as i respond here it takes about 800ms. I'm on a 30Mbps connection in India,

<img width="966" alt="screenshot 2018-04-04 23 58 30" src="https://user-images.githubusercontent.com/604117/38326774-482cbd10-3864-11e8-9bc3-f8642d2455a0.png">

**Comment by carols10cents** [comment]
I'm waiting on a review of #1312 which should mitigate the /api/v1/summary slowness. Until that's reviewed and deployed, no other reports are necessary. Thanks!

**Comment by sgrif** [comment]
The problem ultimately boiled down to some bad crawlers which were putting an unreasonable amount of stress on our database server. We've taken action to restrict them and have been seeing our API response times consistently remain under 300ms.

This is generally where we expect things to stay, so please open a new issue if you see things consistently take >1s to load.

**Comment by ashleygwilliams** [comment]
p.s. http://ping.crates.io/ should help us keep up with how long it might take
